packages/meditrak-server/src/database/models/Survey.js

```diff
@@ -66,7 +82,15 @@ const onChangeUpdateDataGroup = async (
   switch (changeType) {
     case 'update': {
       const { code, data_source_id: dataSourceId } = newRecord;
-      return models.dataSource.updateById(dataSourceId, { code });
+      await models.dataSource.updateById(dataSourceId, { code });
+
+      if (newRecord.period_granularity !== oldRecord.period_granularity) {
+        // the `outdated` status of survey responses must be re-calculated, since it depends on
+        // the period granularity of the survey they belong to
+        // no need to await for the operation, just get it going
+        models.surveyResponse.markAsChanged({ survey_id: newRecord.id });
```

Comment:
Using `markAsChanged` here is not optimal, because a record could be changed for any reason (see eg [AggregateDataPusher.js#L398](https://github.com/beyondessential/tupaia/blob/dev/packages/meditrak-server/src/dhis/pushers/data/aggregate/AggregateDataPusher.js#L398), where we just want to update the dhis sync queue). A potential improvement would be being able to add custom metadata in the marked change, which could be used by the change handler to decide whether the change is relevant.